gateway/src/apicast/policy/nginx_filters/nginx_filters.lua

```diff
@@ -0,0 +1,46 @@
+local _M  = require('apicast.policy').new('Logging Policy', 'builtin')
+local new = _M.new
+local get_headers = ngx.req.get_headers
+
+function _M.new(config)
+  local self = new(config)
+  self.headers = {}
+  self.validate = false
+  for _,val in pairs(config.headers or {}) do
+    self.headers[val.name] = val.append
+    self.validate = true
+  end
+  return self
+end
+
+function _M:rewrite(context)
+  if not self.validate then
+    return
+  end
+
+  local headers = get_headers()
+  context.nginx_filters_headers = {}
+
+  for key,val in pairs(self.headers) do
```

Comment:
I think this would look better with something like `header_name, append`. Then in the `if` below you would had `if append` which would make this easier to understand, in my opinion.